src/platform/FailSafeContext.cpp

```diff
@@ -65,9 +82,89 @@ CHIP_ERROR FailSafeContext::DisarmFailSafe()
     mFailSafeArmed                  = false;
     mAddNocCommandHasBeenInvoked    = false;
     mUpdateNocCommandHasBeenInvoked = false;
+    ConfigurationMgr().SetFailSafeArmed(false);
     DeviceLayer::SystemLayer().CancelTimer(HandleArmFailSafe, this);
+
+    if (DeleteFromStorage() != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to delete the captured FailSafeContext from persistent storage");
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void FailSafeContext::SetAddNocCommandInvoked(FabricIndex nocFabricIndex)
+{
+    mAddNocCommandHasBeenInvoked = true;
+    mFabricIndex                 = nocFabricIndex;
+
+    if (CommitToStorage() != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to commit the AddNocCommandHasBeenInvoked of FailSafeContext to persistent storage");
```

Comment:
This seems like a problem, because it means we will not do cleanup properly, leading to security bugs....  Sounds like this needs to stop being a void function and consumers need to fail out on error here.